apidef/oas/default_test.go

```diff
@@ -0,0 +1,284 @@
+				State: State{
+					Active: true,
+				},
+			},
+		}
+
+		assert.Equal(t, expectedTykExtension, *oasDef.GetTykExtension())
+	})
+
+	t.Run("do not override existing tyk extension by default", func(t *testing.T) {
+		oasDef := OAS{
+			T: openapi3.T{
+				Info: &openapi3.Info{
+					Title: "OAS API",
+				},
+				Servers: openapi3.Servers{
+					{
+						URL: "https://example-org.com/api",
+					},
+				},
+			},
+		}
+
+		existingTykExtension := XTykAPIGateway{
+			Info: Info{
+				Name: "New OAS API",
+			},
+			Server: Server{
+				ListenPath: ListenPath{
+					Value: "/new-listen-path",
+				},
+			},
+		}
+
+		oasDef.SetTykExtension(&existingTykExtension)
+
+		err := oasDef.BuildDefaultTykExtension(TykExtensionConfigParams{})
+		assert.Nil(t, err)
+
+		expectedTykExtension := XTykAPIGateway{
+			Server: Server{
+				ListenPath: ListenPath{
+					Value: "/new-listen-path",
+				},
+			},
+			Upstream: Upstream{
+				URL: "https://example-org.com/api",
+			},
+			Info: Info{
+				Name: "New OAS API",
+				State: State{
+					Active: true,
+				},
+			},
+		}
+
+		assert.Equal(t, expectedTykExtension, *oasDef.GetTykExtension())
+	})
+
+	t.Run("override existing tyk extension with supplied params", func(t *testing.T) {
+		oasDef := OAS{
+			T: openapi3.T{
+				Info: &openapi3.Info{
+					Title: "OAS API",
+				},
+				Servers: openapi3.Servers{
+					{
+						URL: "https://example-org.com/api",
+					},
+				},
+			},
+		}
+
+		existingTykExtension := XTykAPIGateway{
+			Info: Info{
+				Name: "New OAS API",
+			},
+			Server: Server{
+				ListenPath: ListenPath{
+					Value: "/listen-api",
+				},
+			},
+		}
+
+		oasDef.SetTykExtension(&existingTykExtension)
+
+		err := oasDef.BuildDefaultTykExtension(TykExtensionConfigParams{
+			ListenPath:  "/new-listen-api",
+			UpstreamURL: "https://example.org/api",
+		})
+
+		assert.Nil(t, err)
+
+		expectedTykExtension := XTykAPIGateway{
+			Server: Server{
+				ListenPath: ListenPath{
+					Value: "/new-listen-api",
+				},
+			},
+			Upstream: Upstream{
+				URL: "https://example.org/api",
+			},
+			Info: Info{
+				Name: "New OAS API",
+				State: State{
+					Active: true,
+				},
+			},
+		}
+
+		assert.Equal(t, expectedTykExtension, *oasDef.GetTykExtension())
+	})
+
+	t.Run("error when supplied invalid upstreamURL param", func(t *testing.T) {
+		oasDef := OAS{
+			T: openapi3.T{
+				Info: &openapi3.Info{
+					Title: "OAS API",
+				},
+				Servers: openapi3.Servers{
+					{
+						URL: "https://example-org.com/api",
+					},
+				},
+			},
+		}
+
+		existingTykExtension := XTykAPIGateway{
+			Info: Info{
+				Name: "New OAS API",
+			},
+			Server: Server{
+				ListenPath: ListenPath{
+					Value: "/listen-api",
+				},
+			},
+		}
+
+		oasDef.SetTykExtension(&existingTykExtension)
+
+		err := oasDef.BuildDefaultTykExtension(TykExtensionConfigParams{
+			ListenPath:  "/new-listen-api",
+			UpstreamURL: "invalid-url",
+		})
+		assert.EqualError(t, err, "invalid upstream URL")
```

Comment:
This should always be avoided. If you need/want to compare errors like this, define an ErrInvalidUpstreamURL and compare that with assert.ErrorIs